### ISSUE TITLE
Kcctl doesnt work on new M1 Macs

### ISSUE THREAD

**Issue by sfc-gh-japatel** [kcctl/kcctl#245]
kcctl doesnt work on M1 macs, any tricks to make it work?

**Comment by sfc-gh-japatel** [comment]
@sfc-gh-rcheng

**Comment by gunnarmorling** [comment]
Interesting, I would expect this to work actually, by means of the Rosetta emulation which whould should run the x86-64 binary as-is? In general it would be nice though indeed to have native Aarch64 binaries for M1/M2 too, but last time I checked, there was no way for building those via GH Actions. Not sure whether this has changed since then.

**Comment by C0urante** [comment]
I was able to install and run the [1.0.0 early access release](https://github.com/kcctl/kcctl/releases/tag/1.0.0-early-access) on an M1 Mac with Rosetta by downloading the `kcctl-1.0.0-early-access-osx-x86_64.zip` artifact, removing the quarantine flag (as [described in the README](https://github.com/kcctl/kcctl/#installation)), then executing `bin/kcctl` in the unzipped directory.

TL;DR: It works out of the box on my (M1+Rosetta) machine.

**Comment by gunnarmorling** [comment]
Yes, that's observation, too. @sfc-gh-japatel, is this still an issue for you guys, or can we close this one?

**Comment by barthand** [comment]
This might be a wrong place to report, but it fails for me to do a fresh installation of `kcctl` on M1 machine, neither using Homebrew nor SDKMAN.

Homebrew:
```
$ brew install kcctl/tap/kcctl                                                                                                                                                                       

Error: formulae require at least a URL
/opt/homebrew/Library/Homebrew/formula.rb:282:in `determine_active_spec'
/opt/homebrew/Library/Homebrew/formula.rb:213:in `initialize'
/opt/homebrew/Library/Homebrew/formulary.rb:308:in `new'
/opt/homebrew/Library/Homebrew/formulary.rb:308:in `get_formula'
/opt/homebrew/Library/Homebrew/formulary.rb:469:in `get_formula'
/opt/homebrew/Library/Homebrew/formulary.rb:561:in `factory'
/opt/homebrew/Library/Homebrew/cli/parser.rb:633:in `block in formulae'
/opt/homebrew/Library/Homebrew/cli/parser.rb:629:in `map'
/opt/homebrew/Library/Homebrew/cli/parser.rb:629:in `formulae'
/opt/homebrew/Library/Homebrew/cli/parser.rb:307:in `parse'
/opt/homebrew/Library/Homebrew/cmd/install.rb:149:in `install'
/opt/homebrew/Library/Homebrew/brew.rb:95:in `<main>'
```

SDKMan (preceded with `sdk update`):
```
$ sdk install kcctl                                                                                                                                                                                  

Stop! kcctl 1.0.0.Alpha5 is not available. Possible causes:
 * 1.0.0.Alpha5 is an invalid version
 * kcctl binaries are incompatible with your platform
 * kcctl has not been released yet

Tip: see all available versions for your platform:

  $ sdk list kcctl
```

This might give user an impression that `kcctl` doesn't run on Apple Silicon, which is not true.

I suppose Homebrew part is tracked in #134, please let me know if I should report SDKMAN/any other part separately.

**Comment by sebastienblanc** [comment]
> This might be a wrong place to report, but it fails for me to do a fresh installation of `kcctl` on M1 machine, neither using Homebrew nor SDKMAN.
> 
> Homebrew:
> 
> ```
> $ brew install kcctl/tap/kcctl                                                                                                                                                                       
> 
> Error: formulae require at least a URL
> /opt/homebrew/Library/Homebrew/formula.rb:282:in `determine_active_spec'
> /opt/homebrew/Library/Homebrew/formula.rb:213:in `initialize'
> /opt/homebrew/Library/Homebrew/formulary.rb:308:in `new'
> /opt/homebrew/Library/Homebrew/formulary.rb:308:in `get_formula'
> /opt/homebrew/Library/Homebrew/formulary.rb:469:in `get_formula'
> /opt/homebrew/Library/Homebrew/formulary.rb:561:in `factory'
> /opt/homebrew/Library/Homebrew/cli/parser.rb:633:in `block in formulae'
> /opt/homebrew/Library/Homebrew/cli/parser.rb:629:in `map'
> /opt/homebrew/Library/Homebrew/cli/parser.rb:629:in `formulae'
> /opt/homebrew/Library/Homebrew/cli/parser.rb:307:in `parse'
> /opt/homebrew/Library/Homebrew/cmd/install.rb:149:in `install'
> /opt/homebrew/Library/Homebrew/brew.rb:95:in `<main>'
> ```
> 
> SDKMan (preceded with `sdk update`):
> 
> ```
> $ sdk install kcctl                                                                                                                                                                                  
> 
> Stop! kcctl 1.0.0.Alpha5 is not available. Possible causes:
>  * 1.0.0.Alpha5 is an invalid version
>  * kcctl binaries are incompatible with your platform
>  * kcctl has not been released yet
> 
> Tip: see all available versions for your platform:
> 
>   $ sdk list kcctl
> ```
> 
> This might give user an impression that `kcctl` doesn't run on Apple Silicon, which is not true.
> 
> I suppose Homebrew part is tracked in #134, please let me know if I should report SDKMAN/any other part separately.

I ran exactly into the same issue and by googling found this GH issue :)

**Comment by antonmry** [comment]
I installed it in my Mac m1 with sdkman using a rosetta terminal. This is the process in case it can help others: https://apple.stackexchange.com/questions/428768/on-apple-m1-with-rosetta-how-to-open-entire-terminal-iterm-in-x86-64-architec

Once installed, it can be used in a regular terminal (I assume it's slower but it works well enough).

It would be good to add some warning about it in the README

**Comment by gunnarmorling** [comment]
So I think this all should be resolved now. Could someone try with today's release of 1.0.0.Beta3 and let us know whether this can be installed flawlessly? Thanks!

**Comment by gunnarmorling** [comment]
That's officially done now; we publish native aarch64 binaries for macOS as of yesterday's release of 1.0.0.CR3.
